corehq/motech/fhir/tests/test_api_endpoints.py

```diff
@@ -0,0 +1,290 @@
+BAZ_CASE_ID = uuid4().hex
+
+
+@skip('Covered functionality not yet implemented')
+class PatientEndpointTests(TestCase, DomainSubscriptionMixin):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.domain_obj = create_domain(DOMAIN)
+        # Assumes FHIR API needs Pro plan
+        cls.setup_subscription(DOMAIN, SoftwarePlanEdition.PRO)
+        cls.user = WebUser.create(DOMAIN, API_USERNAME, API_PASSWORD,
+                                  created_by=None, created_via=None)
+        create_person_cases(owner_id=cls.user.user_id)
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.user.delete(deleted_by=None)
+        cls.teardown_subscriptions()
+        cls.domain_obj.delete()
+        clear_plan_version_cache()
+        super().tearDownClass()
+
+    def test_search(self):
+        # Tests filtering by a non-indexed case property
+        url = get_endpoint_url(BASE_URL, '/Patient?birthdate=1990-01-01')
+        response = requests.get(url, auth=(API_USERNAME, API_PASSWORD))
+
+        # Some things will be different between the expected search
+        # result and the actual response. Use `json_diff()` to confirm
+        # expected differences, and that everything else is the same.
+        diffs = sorted([diff.path for diff in json_diff(response.json(),
+                                                        get_search_result())])
+        self.assertEqual(diffs, ['id', 'meta.lastUpdated'])
+
+    def test_get(self):
+        url = get_endpoint_url(BASE_URL, f'/Patient/{FOO_CASE_ID}')
+        response = requests.get(url, auth=(API_USERNAME, API_PASSWORD))
+        self.assertEqual(response.json(), FOO_PATIENT)
+
+    def test_auth_bad_password(self):
+        # Authentication should use the same code as existing API
+        url = get_endpoint_url(BASE_URL, f'/Patient/{FOO_CASE_ID}')
+        with self.assertRaisesRegex(requests.HTTPError, '[Ff]orbidden'):
+            requests.get(url, auth=(API_USERNAME, 'bad_password'))
+
+    def test_auth_bad_username(self):
+        url = get_endpoint_url(BASE_URL, f'/Patient/{FOO_CASE_ID}')
+        with self.assertRaisesRegex(requests.HTTPError, '[Ff]orbidden'):
+            # Error should be the same as for bad password: It should
+            # not reveal that the user does not exist.
+            requests.get(url, auth=('admin@example.com', API_PASSWORD))
+
+    def test_auth_bad_resource(self):
+        url = get_endpoint_url(BASE_URL, f'/Patient/{BAZ_CASE_ID}')
+        with self.assertRaisesRegex(requests.HTTPError, '[Nn]ot found'):
+            requests.get(url, auth=(API_USERNAME, API_PASSWORD))
+
+
+def create_person_cases(owner_id):
+    """
+    Creates cases for Fred Foo and Barbara Bar.
+    """
+    different_owner_id = uuid4().hex
+    submit_case_blocks([
+        get_foo_caseblock(owner_id).as_text(),
+        get_bar_caseblock(owner_id).as_text(),
+        get_baz_caseblock(different_owner_id).as_text(),
```

Comment:
nit: `different_owner_id variable seems redundant.

Reply:
The naming is slightly helpful in making it explicit but i think the question "why a different owner id" is still not answered. Re-generating a uuid would already tell that you want a different id.

Reply:
I merged before reading this. :unamused: Sorry.

This is used by the tests to show that search results only include the resources that the user is allowed to see. If the user specifically asks for a resource they are not allowed to see, they get a 404 Not Found (not a 403 Forbidden, because users should not be made aware of the existence of resources they are not allowed to see).